From 5b92b2280b079b04fd8eea37f82b64d9745b7617 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 10 Jan 2017 22:02:29 -0500
Subject: [PATCH 4/4] Add work-around for "duplicated directive" warning

add_directive() modifies global state in docutils.parsers.rst.directives,
even though it looks like a method on the Sphinx() object. So when
add_directive() is called multiple times, warning is emitted. This might
be an error in add_directive() design, but let's just add this hack for
now. This is only tests after all.
---
 sphinxcontrib/programoutput.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/sphinxcontrib/programoutput.py b/sphinxcontrib/programoutput.py
index 3f6a4f1595..f2a61f0ba0 100644
--- a/sphinxcontrib/programoutput.py
+++ b/sphinxcontrib/programoutput.py
@@ -44,6 +44,7 @@
 
 from docutils import nodes
 from docutils.parsers import rst
+from docutils.parsers.rst import directives
 from docutils.parsers.rst.directives import flag, unchanged, nonnegative_int
 
 
@@ -252,12 +253,13 @@ def init_cache(app):
     if not hasattr(app.env, 'programoutput_cache'):
         app.env.programoutput_cache = ProgramOutputCache()
 
-
 def setup(app):
     app.add_config_value('programoutput_use_ansi', False, 'env')
     app.add_config_value('programoutput_prompt_template',
                          '$ {command}\n{output}', 'env')
-    app.add_directive('program-output', ProgramOutputDirective)
-    app.add_directive('command-output', ProgramOutputDirective)
+    if 'program-output' not in directives._directives:
+        # avoid double registration which causes a problem with sphinx 1.6
+        app.add_directive('program-output', ProgramOutputDirective)
+        app.add_directive('command-output', ProgramOutputDirective)
     app.connect(str('builder-inited'), init_cache)
     app.connect(str('doctree-read'), run_programs)
-- 
2.11.0

